incubator/hnc/e2e/tests/issues_test.go

```diff
@@ -36,4 +42,71 @@ var _ = Describe("Issues", func() {
 		runShouldContain("No conditions", 1, "kubectl hns describe", nsParent)
 		runShouldContain("No conditions", 1, "kubectl hns describe", nsChild)
 	})
+
+	It("Should set SubnamespaceAnchorMissing condition if the anchor is missing - issue #501", func() {
+		// Setting up a 3-level tree with 'parent' as the root
+		mustRun("kubectl create ns", nsParent)
+		// create a subnamespace without anchor by creating a full namespace with SubnamespaceOf annotation 
+		mustRun("kubectl create ns", nsSub1)
+		mustRun("kubectl hns set", nsSub1, "--parent", nsParent)
+		mustRun("kubectl annotate ns", nsSub1, "hnc.x-k8s.io/subnamespaceOf=" + nsParent)
+		// If the subnamespace doesn't allow cascadingDelete and the anchor is missing in the parent namespace, it should have 'SubnamespaceAnchorMissing' condition while its descendants shoudn't have any conditions."
+		// Expected: 'sub1' namespace is not deleted and should have 'SubnamespaceAnchorMissing' condition; no other conditions."
+		runShouldContain("SubnamespaceAnchorMissing", 1, "kubectl get hierarchyconfigurations.hnc.x-k8s.io -n", nsSub1, "-o yaml")
+	})
+
+	It("Should unset SubnamespaceAnchorMissing condition if the anchor is read - issue #501", func(){
```

Comment:
*if the anchor is **readded**

Reply:
What's the difference between read / readded? I thought both of them are past tense

Reply:
Oh it's not read but re-add or you can say added later?

Reply:
Make so much more sense now LOL. Done